overlord/snapstate/storehelpers.go

```diff
@@ -69,6 +70,28 @@ func userFromUserID(st *state.State, userIDs ...int) (*auth.UserState, error) {
 	return user, err
 }
 
+func refreshOptions(st *state.State, origOpts *store.RefreshOptions) (*store.RefreshOptions, error) {
+	var opts store.RefreshOptions
+
+	if origOpts != nil {
+		if origOpts.RequestSalt != "" {
+			// nothing to add
+			return origOpts, nil
+		}
+		opts = *origOpts
+	}
+
+	if err := st.Get("refresh-request-salt", &opts.RequestSalt); err != nil && err != state.ErrNoState {
+		return nil, fmt.Errorf("cannot obtain store request salt: %v", err)
+	}
+	if opts.RequestSalt == "" {
```

Comment:
Would it make sense to check the state on startup and add the salt then? Maybe even as a patch?

Reply:
I've discussed this with @mvo5 and @pedronis. The current solution is to use `snapstate.Manager()` to fill request salt.